backstage/packages/app/src/App.tsx

```diff
@@ -39,6 +44,22 @@ import { searchPage } from './components/search/SearchPage';
 import { PrefixNavigate } from './components/utils/PrefixNavigate';
 
 const app = createApp({
+  components: {
+    SignInPage: props => (
+      <SignInPage
+        {...props}
+        providers={[
+          'guest',
+          {
+            id: 'github-auth-provider',
+            title: 'GitHub',
+            message: 'Sign in using GitHub',
+            apiRef: githubAuthApiRef,
+          },
+        ]}
+      />
+    ),
```

Comment:
Am I correct in thinking this shows up before the user is able to perform any other action? If so, can we somehow make `guest` the default and only require additional sign-in when a privileged action is requested (e.g. component template creation)?

Reply:
Yes it does, you have to click Enter as Guest user, when first launched. I don't know if that is possible, but I will investigate.

Reply:
Guests can go to `Settings > Authentication Providers` to login via GitHub.

We might need a [custom auth resolver](https://backstage.io/docs/auth/identity-resolver/#building-custom-resolvers) to automatically login as guest. [This](https://backstage.io/docs/auth/identity-resolver/#quick-start) might also be a problem:

> This provider makes all users share a single "guest" identity. This is useful for testing purposes and quickly getting started locally, but is not safe for use in production and that particular provider will refuse to work there.

Reply:
The shared guest identity isn't an issue for us as we run backstage in an immutable configuration. If you were to allow people to edit things in the UI it would certainly be an issue